src/OpenTelemetry.Contrib.Instrumentation.Runtime/Implementation/GcInstrumentation.cs

```diff
@@ -0,0 +1,102 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Metrics;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Runtime.Implementation
+{
+    internal class GcInstrumentation : IRuntimeInstrumentation, IDisposable
+    {
+        private const string CommittedCounterName = "gc-committed";
+        private const string GcTimeCounterName = "time-in-gc";
+        private const string Gen0SizeCounterName = "gen-0-size";
+        private const string Gen1SizeCounterName = "gen-1-size";
+        private const string Gen2SizeCounterName = "gen-2-size";
+        private const string LohSizeCounterName = "loh-size";
+        private const string PohSizeCounterName = "poh-size";
+
+        private readonly Meter meter;
+        private readonly IEventCounterStore eventCounterStore;
+        private readonly ObservableCounter<double> gcHeapSizeCounter;
+        private readonly ObservableGauge<int> gen0GCCounter;
+        private readonly ObservableGauge<int> gen1GCCounter;
+        private readonly ObservableGauge<int> gen2GCCounter;
+        private readonly ObservableGauge<long> allocRateCounter;
+        private readonly ObservableCounter<double> fragmentationCounter;
+        private readonly ObservableCounter<double> committedCounter;
+        private readonly ObservableCounter<double> gcTimeCounter;
+        private readonly ObservableCounter<long> gen0SizeCounter;
+        private readonly ObservableCounter<long> gen1SizeCounter;
+        private readonly ObservableCounter<long> gen2SizeCounter;
+        private readonly ObservableCounter<long> lohSizeCounter;
+        private readonly ObservableCounter<long> pohSizeCounter;
+
+        public GcInstrumentation(Meter meter, IEventCounterStore eventCounterStore)
+        {
+            this.meter = meter;
+            this.eventCounterStore = eventCounterStore;
+
+            this.eventCounterStore.Subscribe(GcTimeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(Gen0SizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(Gen1SizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(Gen2SizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(LohSizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(PohSizeCounterName, EventCounterType.Mean);
+
+            this.gcHeapSizeCounter = meter.CreateObservableCounter($"{RuntimeMetrics.MetricPrefix}gc_heap_size", () => (double)(GC.GetTotalMemory(false) / 1_000_000), "MB", "GC Heap Size");
```

Comment:
Can I ask a question?
Why `ObservableCounter` and not `ObservableGauge`? 

From the docs: 
ObservableCounter - ObservableCounter is an Instrument which reports monotonically increasing value(s)
Observable Gauge - ObservableGauge is an asynchronous Instrument which reports non-additive value(s)

The heap size is non-additive value and can decrease.

Reply:
Fixed.